onward/app/weather/controllers/WeatherController.scala

```diff
@@ -1,19 +1,41 @@
 package weather.controllers
 
-import common.{ImplicitControllerExecutionContext, JsonComponent}
+import common.JsonComponent.resultFor
+import common.Seqs.RichSeq
+import common.{Edition, ImplicitControllerExecutionContext, JsonComponent}
 import model.Cached
-import play.api.mvc.{Action, AnyContent, BaseController, ControllerComponents}
-import weather.models.CityId
+import play.api.libs.json.Json.{stringify, toJson}
+import play.api.mvc._
 import weather.WeatherApi
-import common.Seqs._
+import weather.models.{CityId, CityResponse, Weather, WeatherResponse}
 
+import scala.concurrent.Future
 import scala.concurrent.duration._
 
 class WeatherController(weatherApi: WeatherApi, val controllerComponents: ControllerComponents)
     extends BaseController
     with ImplicitControllerExecutionContext {
   val MaximumForecastDays = 10
 
+  def theWeather(): Action[AnyContent] =
+    Action.async { implicit request =>
+      val (country, city, region) = readLocationHeaders
+
+      for {
+        location <- whatIsMyCity(country, city, region)
+        currentWeather <- weatherApi.getWeatherForCityId(CityId(location.id))
+        forecasts <- weatherApi.getForecastForCityId(CityId(location.id))
+      } yield {
+        val weather = Weather(
+          location = location,
+          weather = WeatherResponse.fromAccuweather(currentWeather),
+          forecast = forecasts.map(WeatherResponse.fromAccuweather),
+        )
+        val weatherJson = stringify(toJson(weather))
+        Cached(10.minutes)(resultFor(request, weatherJson))
```

Comment:
We could use `fromWritable` and avoid having to make `resultFor` public. This might help with refactoring as it seems to be the convention used.